src/Portage/Metadata.hs

```diff
@@ -91,39 +95,49 @@ stripGlobalUseFlags m = foldr1 Map.intersection (Map.delete <$> globals <*> [m])
 -- | Pretty print as valid XML a list of flags and their descriptions
 -- from a given 'Map.Map'.
 prettyPrintFlags :: Map.Map String String -> [String]
-prettyPrintFlags m = (\(name,description) ->
-                        "\t\t" ++
-                        (showElement
-                         . add_attr (Attr (blank_name { qName = "name" }) name)
-                         . unode "flag" $ description))
-                     <$> (Map.toAscList . stripGlobalUseFlags $ m)
+prettyPrintFlags m
+  | Map.null m = []
+  | otherwise = ["\t<use>"] ++ go ++ ["\t</use>"]
+  where
+    go = printFlag <$> Map.toAscList (stripGlobalUseFlags m)
+    printFlag (name, description) =
+      "\t\t" ++
+      showElement
+        (add_attr
+          (Attr (blank_name { qName = "name" }) name)
+          (unode "flag" description)
+        )
 
 -- | Pretty print a human-readable list of flags and their descriptions
 -- from a given 'Map.Map'.
 prettyPrintFlagsHuman :: Map.Map String String -> [String]
 prettyPrintFlagsHuman m = (\(name,description) -> A.bold (name ++ ": ") ++
-                            (L.intercalate " " . lines $ description))
+                            (unwords . lines $ description))
                           <$> (Map.toAscList . stripGlobalUseFlags $ m)
-                          
+
 -- | A minimal metadata for use as a fallback value.
-makeMinimalMetadata :: Metadata
-makeMinimalMetadata = Metadata { metadataEmails = ["haskell@gentoo.org"]
-                               , metadataUseFlags = Map.empty
-                               }
+minimalMetadata :: EBuild.EBuild -> Metadata
+minimalMetadata ebuild = mempty
+  { metadataEmails = ["haskell@gentoo.org"]
+  -- TODO: This assumes all pacakges exist on hackage
+  , metadataRemoteIds = S.singleton $ RemoteIdHackage $ EBuild.hackage_name ebuild
```

Comment:
To me, `#2` makes the most sense. What I want to avoid is hackport creating the `<remote-id type="hackage">...` line every time it bumps `hackport-*.ebuild` (for example). It might also make sense to add support for something like this in the ebuild:

```bash
#hackport: flags: -blah
#hackport: no-hackage-remote-id
```

Reply:
[`hackport` is on hackage](https://hackage.haskell.org/package/hackport) :) just like other packages.

But there are some such as `elm` which are not on hackage, and we would have issues here.

First off, on the two commands `merge` and `make-ebuild`. The first depends on network access for querying hackage, but the second does not. I think we should preserve this, as network access should not strictly be needed to parse a .cabal file and translate it to an ebuild. Adding hackage querying here would limit the use of hackport without network access for little gain (that gain being: generating the optional remoteId line in the metadata.xml as a hackage uri rather than the actual upstream source repo location).

My preference is to use the sourcerepo that the package author has provided in the .cabal file. That seems to me to be the most robust solution as we don't need to query hackage (slowing things down when we are using the `make-ebuild` command, or preventing its use entirely) or have different behaviours for different hackport commands (complexity with arguably little gain, and _inconsistency_ of metadata.xml generation depending on which command one uses).

Hackage, I would argue, is more of a package index than an index of source repos. It holds the tarballs and metadata revisions, but actual development happens elsewhere. To me, the remoteId should reflect the genuine upstream location being not hackage but wherever development occurs.

Having said all that, given you're the one doing the work it is entirely up to you! I don't think I have a say over your implementation. But I would argue that in the interests minimising complexity and inconsistency, we use the sourcerepo pointed to in the .cabal file and be done with it.

Reply:
I think there might have been a bit of confusion. I am adding support for making `remote-id` tags from the `source-repository` stanza in the `.cabal` file. However, it also makes sense to add a `<remote-id type="hackage">...</remote-id>` line for any package that is created using `hackport merge` (since this queries the package name from the hackage DB). 

The question is what to do with `hackport make-ebuild`. Do we want to assume the package is on hackage by default? What do we do with packages like elm that are not on hackage? My current opinion is that the best approach is to use a command-line flag such as `--no-hackage-remote-id` to toggle this behavior, and perhaps also a new `#hackage: ...` line to make this behavior persist. I'd like to get this working before merging the PR.

Interesting that hackport is on hackage. I did not know that.

Reply:
Just so I understand: is your work adding two remote-ids for a `hackport merge`? One pointing to hackage, and one pointing to the source repo listed in the .cabal file's source-repository stanza?

Reply:
Yes although `source-repository` isn't guaranteed to be there.